astropy/modeling/parameters.py

```diff
@@ -13,32 +13,46 @@
 import inspect
 import functools
 import numbers
+import operator
 
 import numpy as np
 
+from ..units import Quantity
 from ..utils import isiterable
 from ..utils.compat import ignored
 from ..extern import six
+from .utils import array_repr_oneline
 
 __all__ = ['Parameter', 'InputParameterError']
 
 
-class InputParameterError(ValueError):
+class ParameterError(Exception):
+    """Generic exception class for all exceptions pertaining to Parameters."""
+
+
+class InputParameterError(ValueError, ParameterError):
     """Used for incorrect input parameter values and definitions."""
 
 
+class ParameterDefinitionError(ParameterError):
+    """Exception in declaration of class-level Parameters."""
+
+
 def _tofloat(value):
     """Convert a parameter to float or float array"""
 
     if isiterable(value):
         try:
-            value = np.array(value, dtype=np.float)
+            value = np.asanyarray(value, dtype=np.float)
         except (TypeError, ValueError):
             # catch arrays with strings or user errors like different
             # types of parameters in a parameter set
             raise InputParameterError(
                 "Parameter of {0} could not be converted to "
                 "float".format(type(value)))
+    elif isinstance(value, Quantity):
+        # Quantities are fine as is
```

Comment:
Actually, are they fine? It is unlikely but in principle quanties can be int, so maybe include scalar quantity above
```
if isiterable(value) or isinstance(value, Quantity):
```